Mathlib/CategoryTheory/Galois/Decomposition.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.Galois.Basic
+
+/-!
+# Decomposition of objects into connected components and applications
+
+We show that in a Galois category every object is the (finite) coproduct of connected subobjects.
+This has many useful corollaries, in particular that the fiber of every object
+is represented by a Galois object (TODO).
+
+## Main results
+
+* `has_decomp_connected_components`: Every object is the sum of its (finitely many) connected
+  components.
+* `fiber_in_connected_component`: An element of the fiber of `X` lies in the fiber of some
+  connected component.
+* `connected_component_unique`: Up to isomorphism, for each element `x` in the fiber of `X` there
+  is only one connected component whose fiber contains `x`.
+
+-/
+
+universe u₁ u₂ w
+
+namespace CategoryTheory
+
+open Limits Functor
+
+variable {C : Type u₁} [Category.{u₂} C]
+
+namespace Cofan
```

Comment:
The two declarations you've put in this namespace probably belong in another file.

Reply:
I don't know if it should be a new file or an existing one, but it's too general for this file.

Reply:
I agree, I don't know where to put them though (the reason why I ultimately left them here).

Reply:
Did you try #find_home?

Reply:
Yes, but the result is this file :/ I had some discussions on this with @joelriou here: https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Combine.20coproducts

Ideally one could build a suitably general API for combining cofans, which in theory already exists with the `GradedObject` API. In practice though, using this API amounts to showing some cocone categories are isomorphic in every applied situation which turns out to be longer than just reproving the colimit property (as its done here).

Reply:
I would be fine with creating a new file `CombinedProducts.lean` in the `Limits.Shapes` folder and to collect specialized helpers like this one there. We could also put the one posted in the linked Zulip chat above for Sigma types.

Reply:
"When in doubt, create a new file"

(I just made this up, but it sounds true :grinning:)

Reply:
I created a `CombinedProducts.lean` and added the version for `Fan`s. In a followup PR I can add other similar constructors floating around in the mentioned Zulip stream.

Reply:
One might want to refactor the file structure to move `Products.lean` to `Products/Basic.lean` and move the `CombinedProducts.lean` into that folder. If this is wanted, I can do it, but suggest do it after this PR is merged, to not obfuscate the diff.